resolwe_bio/kb/management/commands/insert_features.py

```diff
@@ -85,9 +87,21 @@ def add_arguments(self, parser):
         parser.add_argument(
             "file_name",
             type=str,
-            help="Tab-separated file with features (supports tab, gz or zip)",
+            help=(
+                "Tab-separated file with features (supports tab, gz or zip)."
+                "When file_name starts with s3:// it is downloaded from S3 "
+                "via boto3 library. See also --region."
+            ),
         )
 
+    def download_from_s3(self, options):
+        """Download files from S3 if necessary and fix the options dict."""
+        match = re.search(r"^s3://([a-z0-9-.]{3,63})/(.*)$", options["file_name"])
+        if match:
+            bucket, key = match.groups()
+            boto3.resource("s3").Bucket(bucket).download_file(key, key)
+            options["file_name"] = key
```

Comment:
I don't like that the input parameter is implicitly changed. It would be much better to only pass the file name to the function, return it and explicitly store it in the parent function.